### PR TITLE
sqlc: migrate from Postgres protocol to libsql-client-rs

### PR DESCRIPTION
**Pull request by psarna**
The new backend speaks HTTP via our libsql-client crate.
Lots of rough edges are still present, e.g. there's no support
for binding parameters yet. However, that was also the case
for Postgres, so I consider this patch a strict improvement.

This series also comes with a minor change to `sqld` (cc @MarinPostma) - `PRAGMA` statements are now recognized and accepted, as DBeaver and other similar tools use them to inspect database state.

### CODE REVIEW

**Review by penberg — APPROVED** [review]

**Comment by psarna** [comment]
(the CI error stemmed from me not updating the url in Ruby tests to localhost:8080)

**Comment by psarna** [comment]
bors merge

File: sqld/src/query_analysis.rs [hunk 1]
```diff
@@ -50,6 +50,7 @@ impl StmtKind {
                 | Stmt::CreateIndex { .. },
             ) => Some(Self::Write),
             Cmd::Stmt(Stmt::Select { .. }) => Some(Self::Read),
+            Cmd::Stmt(Stmt::Pragma { .. }) => Some(Self::Other),
```
Comment (haaawk) [inline sqld/src/query_analysis.rs:53]: Is this safe? As far as I remember HTTP requests are load balanced over a pool of Libsql connections. Isn't it possible for a client to get their pragma statement executed on one of the connections of the pool and the next of their requests being executed on another connection in the pool (which does not have the pragma)? @psarna. CC @MarinPostma

Comment (psarna) [reply sqld/src/query_analysis.rs:53]: It's safe only for readonly pragmas that don't change state (lots of them are like that), we should refuse ones that change the connection state.

Comment (haaawk) [reply sqld/src/query_analysis.rs:53]: Wouldn't it be a bit safer to just allow PRAGMA for Hrana and then give them the category of `Write`?

Comment (psarna) [reply sqld/src/query_analysis.rs:53]: We could just whitelist a few readonly statements and add items to the list if somebody reports an issue

Comment (haaawk) [reply sqld/src/query_analysis.rs:53]: I guess whitelisting makes more sense. The question is whether pragma's should be executed always on primary or for some of them it's ok to be run on the replica?

Comment (MarinPostma) [reply sqld/src/query_analysis.rs:53]: there are two things to consider in my opinion: 
i) pragmas that are readonly and generally safe, should be whitelisted
ii) pragmas that change the state of a connection, which cause 2 problems: 1) how to make sure the replica and master share the same connection state, and 2) in what context should such pragmas be allowed (ok for hrana, and not ok for HTTP for instance)

**Pull request merged**